metron-streaming/Metron-Common/src/test/java/org/apache/metron/bolt/BulkMessageWriterTest.java

```diff
@@ -0,0 +1,147 @@
+    JSONParser parser = new JSONParser();
+    sampleMessage = (JSONObject) parser.parse(sampleMessageString);
+    sampleMessage.put("field", "value1");
+    messageList = new ArrayList<>();
+    messageList.add(((JSONObject) sampleMessage.clone()));
+    sampleMessage.put("field", "value2");
+    messageList.add(((JSONObject) sampleMessage.clone()));
+    sampleMessage.put("field", "value3");
+    messageList.add(((JSONObject) sampleMessage.clone()));
+    sampleMessage.put("field", "value4");
+    messageList.add(((JSONObject) sampleMessage.clone()));
+    sampleMessage.put("field", "value5");
+    messageList.add(((JSONObject) sampleMessage.clone()));
+  }
+
+  @Mock
+  private BulkMessageWriter<JSONObject> bulkMessageWriter;
+
+  @Test
+  public void test() throws Exception {
+    BulkMessageWriterBolt bulkMessageWriterBolt = new BulkMessageWriterBolt("zookeeperUrl").withBulkMessageWriter(bulkMessageWriter);
+    bulkMessageWriterBolt.setCuratorFramework(client);
+    bulkMessageWriterBolt.setTreeCache(cache);
+    bulkMessageWriterBolt.configurations.updateSensorEnrichmentConfig(sensorType, new FileInputStream(sampleSensorEnrichmentConfigPath));
+    bulkMessageWriterBolt.declareOutputFields(declarer);
+    verify(declarer, times(1)).declareStream(eq("error"), argThat(new FieldsMatcher("message")));
+    Map stormConf = new HashMap();
+    doThrow(new Exception()).when(bulkMessageWriter).init(eq(stormConf), any(Configurations.class));
+    try {
+      bulkMessageWriterBolt.prepare(stormConf, topologyContext, outputCollector);
+      fail("A runtime exception should be thrown when bulkMessageWriter.init throws an exception");
+    } catch(RuntimeException e) {}
+    reset(bulkMessageWriter);
+    bulkMessageWriterBolt.prepare(stormConf, topologyContext, outputCollector);
+    verify(bulkMessageWriter, times(1)).init(eq(stormConf), any(Configurations.class));
+    tupleList = new ArrayList<>();
+    for(int i = 0; i < 4; i++) {
+      when(tuple.getValueByField("message")).thenReturn(messageList.get(i));
+      tupleList.add(tuple);
+      bulkMessageWriterBolt.execute(tuple);
+      verify(bulkMessageWriter, times(0)).write(eq(sensorType), any(Configurations.class), eq(tupleList), eq(messageList));
+    }
+    when(tuple.getValueByField("message")).thenReturn(messageList.get(4));
+    tupleList.add(tuple);
+    bulkMessageWriterBolt.execute(tuple);
+    verify(bulkMessageWriter, times(1)).write(eq(sensorType), any(Configurations.class), eq(tupleList), argThat(new MessageListMatcher(messageList)));
+    verify(outputCollector, times(5)).ack(tuple);
+    reset(outputCollector);
+    doThrow(new Exception()).when(bulkMessageWriter).write(eq(sensorType), any(Configurations.class), Matchers.anyListOf(Tuple.class), Matchers.anyListOf(JSONObject.class));
+    when(tuple.getValueByField("message")).thenReturn(messageList.get(0));
+    for(int i = 0; i < 5; i++) bulkMessageWriterBolt.execute(tuple);
```

Comment:
This is a style thing, but can we please have braces?  It's so easy to accidentally screw if's and for's up without explicit braces.